### ISSUE TITLE
 How to know that my data set is well done?

### ISSUE THREAD

**Issue by e-click** [clovaai/deep-text-recognition-benchmark#371]
I am trying to train a model to use it with EasyOCR in an ANPR project, as a first test I used a dataset that is provided in the EasyOCR repository, this is an example of the images used.
![0](https://user-images.githubusercontent.com/105033880/202492671-b50e2d78-31c6-4065-ada5-e8c14fff3aa6.jpg)
The dataset contains approximately 1000 similar images in different sizes, after a training of 10,000 iterations I got poor results but since it was a training test I didn't give it more importance.

After that I tried to train my own model following the same steps I used to train the previous model, create my .txt file with the labels and run the script to create the mdb dataset, but after training I don't get any result, it stays on the same prediction and it never gets better.
Example of images:
![Screenshot 2022-11-17 095720](https://user-images.githubusercontent.com/105033880/202495183-430791b4-3c0f-43cb-bd5a-12e5f8b579c8.png)

Another thing I noticed is that after the training script runs the dataset filter, I only had about 100 samples left out of 500 that the dataset has.

The training seems as if it was stuck since it never achieved a correct prediction, unlike with the first dataset, the training after 2000 iterations already gave good results.

These are my logs and results of failure training.

exp_name: None-VGG-BiLSTM-CTC-Seed1111
train_data: data
valid_data: data/val
manualSeed: 1111
workers: 0
batch_size: 192
num_iter: 10000
valInterval: 100
saved_model: 
FT: False
adam: False
lr: 1
beta1: 0.9
rho: 0.95
eps: 1e-08
grad_clip: 5
baiduCTC: False
select_data: ['train']
batch_ratio: ['1']
total_data_usage_ratio: 1.0
batch_max_length: 25
imgH: 32
imgW: 100
rgb: False
character: 0123456789-abcdefghijklmnopqrstuvwxyz
sensitive: False
PAD: False
data_filtering_off: False
Transformation: None
FeatureExtraction: VGG
SequenceModeling: BiLSTM
Prediction: CTC
num_fiducial: 20
input_channel: 1
output_channel: 512
hidden_size: 256
num_gpu: 0
num_class: 38
---------------------------------------

dataset_root: data
opt.select_data: ['train']
opt.batch_ratio: ['1']
--------------------------------------------------------------------------------
dataset_root:    data	 dataset: train
sub-directory:	/train	 num samples: 102
num total samples of train: 102 x 1.0 (total_data_usage_ratio) = 102
num samples of train per batch: 192 x 1.0 (batch_ratio) = 192
--------------------------------------------------------------------------------
Total_batch_size: 192 = 192
--------------------------------------------------------------------------------
dataset_root:    data/val	 dataset: /
sub-directory:	/.	 num samples: 1
--------------------------------------------------------------------------------

[8400/10000] Train loss: 0.00001, Valid loss: 9.07022, Elapsed_time: 46794.15422
Current_accuracy : 0.000, Current_norm_ED  : 0.14
Best_accuracy    : 0.000, Best_norm_ED     : 0.14
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
4nt5526                   | 2103j6                    | 0.0474	False
--------------------------------------------------------------------------------
[8500/10000] Train loss: 0.00001, Valid loss: 9.07421, Elapsed_time: 47353.40529
Current_accuracy : 0.000, Current_norm_ED  : 0.14
Best_accuracy    : 0.000, Best_norm_ED     : 0.14
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
4nt5526                   | 2103j6                    | 0.0477	False
--------------------------------------------------------------------------------
[8600/10000] Train loss: 0.00001, Valid loss: 9.07817, Elapsed_time: 47907.41048
Current_accuracy : 0.000, Current_norm_ED  : 0.14
Best_accuracy    : 0.000, Best_norm_ED     : 0.14
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
4nt5526                   | 2103j6                    | 0.0479	False
--------------------------------------------------------------------------------
[8700/10000] Train loss: 0.00001, Valid loss: 9.08219, Elapsed_time: 49519.87970
Current_accuracy : 0.000, Current_norm_ED  : 0.14
Best_accuracy    : 0.000, Best_norm_ED     : 0.14
--------------------------------------------------------------------------------
Ground Truth              | Prediction                | Confidence Score & T/F
--------------------------------------------------------------------------------
4nt5526                   | 2103j6                    | 0.0481	False
--------------------------------------------------------------------------------

**Comment by AKKHTN2131999** [comment]
Each training requires 192 images. But you only have 102 photos. Is it too little?
I don't understand why it can run?

**Comment by elechocruz** [comment]
> Each training requires 192 images. But you only have 102 photos. Is it too little? I don't understand why it can run?

Yeah thats one problem, my dataset originally has 500 images, but after the filter only has 102, i don't know what filter does and why filter too much, do you think that it is the reason of the bad training? 
(I'm the OP but with other account)

**Comment by AKKHTN2131999** [comment]
I also have the same question with you. I don't know my dataset have enough good to train. You can print the label in train set to know what filter. 
https://github.com/clovaai/deep-text-recognition-benchmark/blob/68a80fe97943a111ff1efaf52a63ad8f0f1c0e5d/dataset.py#L158

**Comment by AdamKheire** [comment]
@e-click  after deploying Easyocr how about the accuracy of the prediction i'm facing poor performance
